src/controllers/collaboratorController.js

```
@@ -0,0 +1,94 @@
+const collaboratorQueries = require("../db/queries.collaborators.js");
+const Authorizer = require("../policies/application");
+const User = require("../db/models").User;
+const Collaborator = require("../db/models").Collaborator;
+const wikiQueries = require("../db/queries.wikis");
+const userQueries = require("../db/queries.users");
+
+module.exports = {
+    addCollaborator(req, res, next){
+            User.findAll({where: {id: req.body.userId}}).then((users) => {
+                users.forEach((user) => {
+                    const authorized = new Authorizer(user)._isPremium();
+
+                    if(authorized) {
+                        collaboratorQueries.createCollaborator(req.body.userId, req.params.id, (err, collaborator) => {
+                            if(err){
+                                console.log(err);
+                                req.flash("error", err);
+                            } else {
+                                req.flash("notice", "New collaborator addedd successfully!");
+                                res.redirect(`/wikis/${req.params.id}`);
+                            }
+                        })
+                    } else {
+                        console.log("must be premium users");
+                        req.flash("notice", "Only premium users can be added as collaborators");
+                        res.redirect(`/wikis/${req.params.id}/addCollaborator`);
+                    }
+                })
+            })
+            .catch((err) => {
+            console.log(err);
```

swallowing exceptions bad